src/linter.js

```diff
@@ -415,12 +427,42 @@ export default class Linter {
       });
   }
 
+  get shouldScanFile() {
+    if (this.config.scanFile) {
+      const manifestFileNames = [
+        'install.rdf', 'manifest.json', 'package.json',
+      ];
+
+      return (fileOrDirName, isDir=false) => {
```

Comment:
Can we make this a private method in the file somewhere? I just get surprised when I see inline functions returned as first-class objects as we don't do it elsewhere in the code often (if ever).